walkeeper/src/broker.rs

```diff
@@ -0,0 +1,201 @@
+//! Communication with etcd, providing safekeeper peers and pageserver coordination.
+
+use anyhow::bail;
+use anyhow::Context;
+use anyhow::Error;
+use anyhow::Result;
+use etcd_client::Client;
+use etcd_client::EventType;
+use etcd_client::PutOptions;
+use etcd_client::WatchOptions;
+use lazy_static::lazy_static;
+use regex::Regex;
+use serde::{Deserialize, Serialize};
+use serde_with::{serde_as, DisplayFromStr};
+use std::str::FromStr;
+use std::time::Duration;
+use tokio::task::JoinHandle;
+use tokio::{runtime, time::sleep};
+use tracing::*;
+use zenith_utils::zid::ZTenantId;
+use zenith_utils::zid::ZTimelineId;
+use zenith_utils::{
+    lsn::Lsn,
+    zid::{ZNodeId, ZTenantTimelineId},
+};
+
+use crate::{safekeeper::Term, timeline::GlobalTimelines, SafeKeeperConf};
+
+const RETRY_INTERVAL_MSEC: u64 = 1000;
+const PUSH_INTERVAL_MSEC: u64 = 5000;
```

Comment:
May be useful to move these durations to the config, to adjust precisely. Also I'm not sure if values were specially chosen already, 60s TTL seems large to me, I'd prefer 5s TTL with regular keepalives (once a second for example).

Reply:
Made them smaller, can do configurable later if we want.